### PR TITLE
AzureCommunicationChat swiftinterface

### PR DESCRIPTION
**Pull request by knvsl**

### CODE REVIEW

File: examples/x86_64.swiftinterface [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+  public init(from decoder: Swift.Decoder) throws
+  public func encode(to encoder: Swift.Encoder) throws
+}
+public struct UpdateChatThreadRequest : Swift.Codable {
+  public let topic: Swift.String?
+  public init(topic: Swift.String? = nil)
+  public init(from decoder: Swift.Decoder) throws
+  public func encode(to encoder: Swift.Encoder) throws
+}
+public struct AzureCommunicationChatClientOptions : AzureCore.ClientOptions {
```
Comment (knvsl) [inline examples/x86_64.swiftinterface:205]: Copying over from old PR - we could shorten this to ChatClientOptions

**Comment by tjprescott** [comment]
Old PR, for reference: https://github.com/Azure/azure-sdk-for-ios/pull/851
